acpreprocessing/stitching_modules/convert_to_n5/psdeskew.py

```diff
@@ -0,0 +1,147 @@
+        si.append(np.ravel_multi_index((szv,sxv),(sdims[0],sdims[2])))
+    kwargs = {'dsi' : dsi,
+              'si' : si,
+              'slice1d' : np.zeros((subblocks,blockdims[1],blockdims[2]*blockdims[0]),dtype=dtype),
+              'blockdims' : blockdims,
+              'subblocks' : subblocks,
+              'flip' : deskewFlip,
+              'dtype' : dtype,
+              'chunklength': blockx
+              }
+    return kwargs
+        
+def deskew_block(blockData,n,dsi,si,slice1d,blockdims,subblocks,flip,dtype,chunklength,*args,**kwargs):
+    """deskew a data chunk in sequence with prior chunks
+    
+    Parameters
+    ----------
+    blockData : numpy.ndarray
+        block of raw (nondeskewed) data to be deskewed
+    n : int
+        current iteration in block sequence (must be run sequentially)
+    dsi : numpy.ndarray
+        deskewed indices for reslicing flattened data
+    si : numpy.ndarray
+        skewed indices for sampling flattened raw data
+    slice1d : numpy.ndarray
+        flattened data from previous iteration containing data for next deskewed block
+    blockdims : tuple of int
+        dimensions of output block
+    subblocks : int
+        number of partitions of input block for processing - likely not necessary
+    flip : bool
+        deskew flip
+    dtype : str
+        datatype
+    chunklength : int
+        number of slices expected for raw data block (for zero filling)
+    
+    Returns
+    ----------
+    ndarray of pixel shifted deskewed data ordered (z,y,x) by sample axes 
+    """
+    subb = subblocks
+    block3d = np.zeros(blockdims,dtype=dtype)
+    zdim = block3d.shape[0]
+    ydim = block3d.shape[1]
+    xdim = block3d.shape[2]
+    # crop blockData if needed
+    if blockData.shape[1] > ydim:
+        y0 = int(np.floor((blockData.shape[1]-ydim)/2))
+        y1 = int(np.floor((blockData.shape[1]+ydim)/2))
+        blockData = blockData[:,y0:y1,:]
+    #print('deskewing block ' + str(n) + ' with shape ' + str(blockData.shape))
+    if blockData.shape[0] < chunklength:
+        #print('block is short, filling with zeros')
+        blockData = np.concatenate((blockData,np.zeros((int(chunklength-blockData.shape[0]),blockData.shape[1],blockData.shape[2]))))
+    order = (np.arange(subb)+n)%subb
+    for y in range(ydim):
+        for i,o in enumerate(order):
+            # flip stack axis 2 for ispim2
+            s = -1 if flip else 1
+            slice1d[o,y,:][dsi[i]] = blockData[:,y,::s].ravel()[si[i]]
+        block3d[:,y,:] = slice1d[n%subb,y,:].reshape((zdim,xdim))
+        slice1d[n%subb,y,:]=0
+    return block3d
+    
+def reshape_joined_shapes(joined_shapes,blockdims,*args,**kwargs):
+    deskewed_shape = (int(np.ceil(joined_shapes[0]/blockdims[2])*blockdims[2]),
+                                   blockdims[1],
+                                   blockdims[0])
+    return deskewed_shape
+
+def options_from_str(idstr):
```

Comment:
I agree - this is a main issue I wanted to seek guidance on. Is the best way forward to perhaps implement a new argschema class for the deskew options specifically? However for the pixel deskew, it seems like a standard "config" (represented here by the "ispim2" flag) is useful, particularly because the deskew parameters are based on acquisition/scope parameters. Is there a proper way to gather argschema parameters from a config file?

Reply:
there is some open discussion in argschema about ways to expand this and gather configurations from multiple sources, but the use case I understand here is essentially the same as defining a schema and making a base input json to use for ispim2 processing.  e.g. running something with `<myscript> --input_dir <input_dir> --output_dir <output_dir> --input_json ispim2_inputs.json`